generator/standalone.cc

```diff
@@ -0,0 +1,171 @@
+#include <fstream>
+#include <standalone.h>
+
+#include <google/protobuf/compiler/command_line_interface.h>
+#include <google/protobuf/stubs/io_win32.h>
+#include <gapic_generator.h>
+
+namespace google {
+namespace api {
+namespace codegen {
+
+namespace pb = google::protobuf;
+
+// GAPIC Standalone expects proto package as an argument. It defines the API
+// for which the GAPIC code must be generated for (i.e. to distinguish between
+// common proto descriptors and the actual API descriptors). Protobuf compiler
+// and the corresponding google::protobuf::CommandLineInterface expect the list
+// of the API-specific .proto files instead.
+//
+// This function extracts the relevant file names (those, which match the
+// package) from the set of descriptors.
+bool ExtractFileNames(const std::vector<std::string>& desc_sets,
+                      const std::vector<std::string>& packages,
+                      std::vector<std::string>& filenames,
+                      std::string* error_msg) {
+  for (const std::string& desc_set : desc_sets) {
+    if (desc_set.empty()) {
+      continue;
+    }
+
+    std::ifstream fstr(desc_set);
+    if (!fstr) {
+      *error_msg = "Could not open file " + desc_set;
+      return false;
+    }
+
+    pb::FileDescriptorSet bin_desc_set;
+    bool parse_result = bin_desc_set.ParseFromIstream(&fstr);
+
+    fstr.close();
+    if (!parse_result) {
+      return false;
+    }
+
+    for (int i = 0; i < bin_desc_set.file_size(); i++) {
+      const auto& fl = bin_desc_set.file(i);
+      auto iter = std::find(packages.begin(), packages.end(), fl.package());
+      if (iter != packages.end()) {
+        filenames.emplace_back(fl.name());
+      }
+    }
+  }
+
+  return true;
+}
+
+void ParseDelimitedArg(const std::string& arg,
```

Comment:
I was trying to avoid dependency on absl for production. I saw it was used in unit tests and assumed it was only in tests, but now I see that gapic-generator-cpp uses it in other places as well. Since we need to split by 2 characters, not one, or not split at all (if there is no splitter), the logic is less trivial than just calling `absl::StrSplit`, I belive it is ok to keep it as is.

Reply:
Hmm, I'm not totally opposed I guess, but what about this code:
```c++
std::vector<std::string> ParseDelimitedArg(const std::string& arg) {
  if (arg.find(':') != std::string::npos) {
    return absl::StrSplit(arg, ':');
  } else {
    return absl::StrSplit(arg, ';');
  }
}
```

Reply:
Or even better, who needs a function?
```c++
absl::StrSplit(arg, absl::ByAnyChar(":;"));
```

See https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_split.h#L174

Reply:
Or, \<value\>\<codegolf\>
```
std::vector<std::string> ParseDelimitedArg(std::string const& arg) {
  char delim = (arg.find(':') != std::string::nps) ? ':' : ';';
  return absl::StrSplit(arg, delim);
}
```
\</codegolf>\</value\>

Reply:
Ok, ok... I wanted to have a joy of pointer arithmetic (haven't done it for years), but you guys want it to be like in any other language these days...

Reply:
@michaelbausor `ByAnyChar()` will change the behavior slightly (it will allow simultaneous usage of  ':' and ';' in same string, which was not allowed initially. But whatever, it is not rally that important.